src/oom/oomd-util.c

```diff
@@ -251,10 +263,18 @@ int oomd_kill_by_swap_usage(Hashmap *h, bool dry_run) {
                         continue;
```

Comment:
Oh, the loop termination is borked! `r` is assigned in the loop, possibly to negative, but also used here.